sdk/core/System.ClientModel/src/Options/ClientPipelineOptions.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Controls the creation of a pipeline used by a service client.
+/// Service clients must create a client-specific subtype of this class
+/// to pass to their constructors to allow for service-specific options
+/// with a client-wide scope.
+/// </summary>
+public class ClientPipelineOptions
+{
+    private static readonly ClientPipelineOptions _defaultOptions = new();
+    internal static ClientPipelineOptions Default => _defaultOptions;
+
+    #region Pipeline creation: Overrides of default pipeline policies
+
+    public PipelinePolicy? RetryPolicy { get; set; }
+
+    public PipelineTransport? Transport { get; set; }
+
+    #endregion
+
+    #region Pipeline creation: Policy settings
+
+    public TimeSpan? NetworkTimeout { get; set; }
+
+    #endregion
+
+    #region Pipeline creation: User-specified policies
+
+    internal PipelinePolicy[]? PerCallPolicies { get; set; }
+
+    internal PipelinePolicy[]? PerTryPolicies { get; set; }
+
+    internal PipelinePolicy[]? BeforeTransportPolicies { get; set; }
+
+    public void AddPolicy(PipelinePolicy policy, PipelinePosition position)
+    {
+        Argument.AssertNotNull(policy, nameof(policy));
+
+        switch (position)
+        {
+            case PipelinePosition.PerCall:
+                PerCallPolicies = AddPolicy(policy, PerCallPolicies);
+                break;
+            case PipelinePosition.PerTry:
+                PerTryPolicies = AddPolicy(policy, PerTryPolicies);
+                break;
+            case PipelinePosition.BeforeTransport:
+                BeforeTransportPolicies = AddPolicy(policy, BeforeTransportPolicies);
```

Comment:
This doesn't appear to be thread safe, since we are replacing the value of the respective policy array. Should we lock here?

Reply:
Isn't this up to the user to call this method? Seems like they should avoid calling it in parallel.

Reply:
Client methods are thread-safe. Models/options are not.

Reply:
given this is only a config time operation, it seems harmless to make it thread safe though.

Reply:
I don't feel super strongly either way on this - I agree models don't need to be threadsafe - do we have a policy for options types?  It's also not a high-performance area of the code so taking a lock shouldn't be a huge problem.  Any thoughts on how best to break the tie?

Reply:
Since I was discussing the other topics with @KrzysztofCwalina offline I got his quick opinion on this one to see if there was a clear answer one way or another -- he said this is an advanced enough type that people should use it correctly, but if they are using it across multiple clients, there is another concern that they might pass it to one client, then modify it and pass it to a different client, and the two clients would behave differently.  Given this, I'll move forward with making this options type freezable (like our [RequestOptions](https://github.com/Azure/azure-sdk-for-net/blob/feature/core-experiment/sdk/core/System.ClientModel/src/Options/RequestOptions.cs#L16) type is), and leave it non-threadsafe.

Shout if either of you have strong concerns about this decision, @christothes or @JoshLove-msft!

Reply:
Tracking with https://github.com/Azure/azure-sdk-for-net/issues/41200